### PR TITLE
Add option to show JsonInput in fullscreen maximised popout

### PR DESCRIPTION
**Pull request by TomTirapani**
Also adds button to JsonInput to auto-format content.
Addresses issue  #1228

Hoist P/R Checklist
-------------------
Review and check off the below. Items that do not apply should be checked to indicate they have been considered. If unclear if a step is relevant, please leave unchecked and note in comments.

- [x] Up to date with `develop` branch as of last change.
- [x] Ready for review (or add `wip` label).
- [x] Added CHANGELOG entry (or N/A)
- [x] Checked for breaking changes (add `breaking-change` label + CHANGELOG if so, or N/A)
- [x] Updated doc comments / prop-types (or N/A)
- [x] Reviewed and tested on Mobile (or N/A)
- [x] Created Toolbox branch / PR (link provided here, or N/A)

### CODE REVIEW

File: desktop/cmp/input/JsonInput.js [hunk 1]
```diff
@@ -47,11 +52,18 @@ export class JsonInput extends HoistInput {
...
-        editorProps: PT.object
+        editorProps: PT.object,
+
+        /** True to show Fullscreen + Auto-format buttons at top-right of input. */
+        showActionButtons: PT.bool
     };
 
     get commitOnChange() {
-        withDefault(this.props.commitOnChange, false);
+        return withDefault(this.props.commitOnChange, false);
```
Comment (amcclain) [inline desktop/cmp/input/JsonInput.js]: Good catch!

File: desktop/cmp/input/JsonInput.scss [hunk 2]
```diff
@@ -29,6 +29,25 @@
       box-sizing: content-box !important;
     }
   }
+
+  &__action-buttons {
+    position: absolute !important;
+    top: 0;
+    right: 6px;
+    z-index: 3;
```
Comment (amcclain) [inline desktop/cmp/input/JsonInput.scss]: Suggest a comment on how we ended up at 3 here. 😄 

I see this works well when e.g. launched from a dialog. The rules for z-index always escape me - does this end up being relative in some sense (inside a "stacking context")?

Comment (TomTirapani) [reply desktop/cmp/input/JsonInput.scss]: Simply put, that's there to place it "above" the code-mirror row and enable clicking on it. The code-mirror styles use `z-index: 2` internally for the rows of content - without this rule, clicking on the buttons selects the first row of content instead.

Just added a comment to the code to clarify.

**Comment by lbwexler** [comment]
👍 Thanks Tom!

**Pull request merged**